### PR TITLE
Refs [TASK][26179]: Create ui SettingFragment

### PR DESCRIPTION
**Pull request by aug16vcc**
## Related Tickets
- [#Ticket ID](https://edu-redmine.sun-asterisk.vn/issues/26179)

## WHAT
- Create ui SettingFragment
## HOW
- Used PreferenceFragmentCompat

## WHY
- in the tasks list

## Evidence (Screenshot or Video)
Screenshot_20200807-080910

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Vu Chi Cong | Reviewer2 (HuyTQ)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/fragment_settings.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <FrameLayout
+        android:id="@android:id/list_container"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_settings.xml:8]: camel case nhé

File: app/src/main/java/com/sunasterisk/boringweather/util/ContextExtensions.kt [hunk 2]
```diff
@@ -4,13 +4,15 @@ import android.app.Notification
...
 import android.content.Context
+import android.content.SharedPreferences
 import android.os.Build
 import android.view.View
 import android.view.inputmethod.InputMethodManager
 import android.widget.Toast
 import androidx.annotation.StringRes
 import androidx.core.app.NotificationCompat
 import androidx.core.content.ContextCompat
+import androidx.preference.PreferenceManager
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/util/ContextExtensions.kt]: sao lại thêm 2 cái import này nhỉ,có thấy thêm code nào đâu ta

File: app/src/main/res/layout/fragment_settings.xml (hunk 1 above)
Comment (aug16vcc) [reply app/src/main/res/layout/fragment_settings.xml:8]: Cái fragment_settings này không được ông ạ. Tại vì đây là id của android cho cái PreferenceFragment ý. Nó fix cứng trong thư viện rồi

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt [hunk 3]
```diff
@@ -123,12 +128,25 @@ class CurrentFragment : BaseFragment(), CurrentContract.View,
...
+            pendingDrawerCloseRunnable = Runnable { findNavigator()?.navigateToSettingsFragment() }
             true
         }
         else -> false
+    }.also { if (it) drawerLayout.closeDrawers() }
+
+    override fun onDrawerClosed(drawerView: View) {
+        pendingDrawerCloseRunnable?.let {
+            it.run()
+            pendingDrawerCloseRunnable = null
```
Comment (tuanpa-0908) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt]:
```suggestion
        pendingDrawerCloseRunnable?.run()
        pendingDrawerCloseRunnable = null
```

File: app/src/main/java/com/sunasterisk/boringweather/ui/setting/SettingsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import com.sunasterisk.boringweather.ui.main.findNavigator
+import kotlinx.android.synthetic.main.fragment_settings.*
+
+class SettingsFragment : PreferenceFragmentCompat() {
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.settings_preferences, rootKey)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```
Comment (tuanpa-0908) [inline app/src/main/java/com/sunasterisk/boringweather/ui/setting/SettingsFragment.kt:16]: Hình như callback này chạy trước, ô để lên trên nhé

Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/boringweather/ui/setting/SettingsFragment.kt:16]: Không ông ạ. `onCreatePreferences` được gọi trong `onCreate` của fragment

**Review by huytq-2422 — APPROVED** [review]

**Pull request merged**